xbmc/windowing/wayland/OSScreenSaverWebOS.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+*  Copyright (C) 2023 Team Kodi
+*  This file is part of Kodi - https://kodi.tv
+*
+*  SPDX-License-Identifier: GPL-2.0-or-later
+*  See LICENSES/README.md for more information.
+*/
+
+#include "OSScreenSaverWebOS.h"
+
+#include "CompileInfo.h"
+#include "utils/JSONVariantParser.h"
+#include "utils/JSONVariantWriter.h"
+#include "utils/log.h"
+
+namespace
+{
+constexpr const char* LUNA_REGISTER_SCREENSAVER =
+    "luna://com.webos.service.tvpower/power/registerScreenSaverRequest";
+constexpr const char* LUNA_RESPONSE_SCREENSAVER =
+    "luna://com.webos.service.tvpower/power/responseScreenSaverRequest";
+} // namespace
+
+namespace KODI::WINDOWING::WAYLAND
+{
+
+COSScreenSaverWebOS::~COSScreenSaverWebOS()
+{
+  if (m_requestContext)
+  {
+    if (HUnregisterServiceCallback(m_requestContext.get()))
+      CLog::LogF(LOGWARNING, "COSScreenSaverWebOS: Luna request unregister failed");
+    m_requestContext = nullptr;
+  }
+}
+
+void COSScreenSaverWebOS::Inhibit()
+{
+  CVariant request;
+  request["subscribe"] = true;
+  request["clientName"] = CCompileInfo::GetPackage();
+  std::string payload;
+  CJSONVariantWriter::Write(request, payload, true);
+
+  m_requestContext = std::make_unique<HContext>();
+  m_requestContext->pub = true;
+  m_requestContext->multiple = true;
+  m_requestContext->callback = &ScreenSaverResponse;
+  if (HLunaServiceCall(LUNA_REGISTER_SCREENSAVER, payload.c_str(), m_requestContext.get()))
+  {
+    CLog::LogF(LOGWARNING, "COSScreenSaverWebOS: Luna request call failed");
+    if (HUnregisterServiceCallback(m_requestContext.get()))
+      CLog::LogF(LOGWARNING, "COSScreenSaverWebOS: Luna request unregister failed");
+    m_requestContext = nullptr;
+  }
+}
+
+void COSScreenSaverWebOS::Uninhibit()
+{
+  if (m_requestContext)
+  {
+    if (HUnregisterServiceCallback(m_requestContext.get()))
+      CLog::LogF(LOGWARNING, "COSScreenSaverWebOS: Luna request unregister failed");
+    m_requestContext = nullptr;
+  }
+}
+
+bool COSScreenSaverWebOS::ScreenSaverResponse(LSHandle* sh, LSMessage* reply, void* ctx)

```

Comment:
the method name is a tad unintuitive,  given that it returns true if the screensaver is inactive.
Maybe ScreenSaverInhibitCallback?

Reply:
Changed to `OnScreenSaverAboutToStart` to indicate that the function is called before the screensaver would be enabled so we can prevent that from happening. It returns `true` because the message was handled.